apps/gallery/js/info.js

```diff
@@ -44,18 +52,72 @@ function showFileInformation(fileinfo) {
           args: args
         },
         'info-type': {raw: fileinfo.type},
-        'info-date': {raw: MediaUtils.formatDate(fileinfo.date)},
+        'info-date': {raw: formattedDate},
         'info-resolution': {
           raw: fileinfo.metadata.width + 'x' + fileinfo.metadata.height
-        }
+        },
       };
+      // placeholders for Exif
+      exifFields.forEach((elem) => {
+        data[elem] = {raw: ''};
+      });
 
       // Populate info overlay view
       MediaUtils.populateMediaInfo(data);
+
+      LazyLoader.load(['shared/js/media/jpeg-exif.js'], () => {
+        photodb.getFile(fileinfo.name, (imagefile) => {
+          JPEGParser.readExifMetaData(imagefile, (error, metadata) => {
+            if (!metadata) {
+              // hide the exif fields.
+              exifFields.forEach((elem) => {
+                setFieldVisibility(elem, false);
+              });
+              return;
+            }
+
+            var data = {};
+            // We don't seem to supply FNumber
+            var fnumber = metadata.FNumber ?
+                (metadata.FNumber.numerator / metadata.FNumber.denominator) :
+                Math.pow(2, (metadata.ApertureValue.numerator /
+                             metadata.ApertureValue.denominator) / 2);
+            // use Exif date if available
+            if (metadata.DateTimeOriginal) {
+              formattedDate = MediaUtils.formatExifDate(
+                metadata.DateTimeOriginal);
+            }
+            data['info-date'] = { raw: formattedDate };
+
+            data['info-aperture'] =  { raw: 'f/' + fnumber.toFixed(1) };
+            data['info-shutterspeed'] =  {
+              raw: metadata.ExposureTime.numerator + '/' +
+                metadata.ExposureTime.denominator
+            };
+            data['info-iso'] = { raw: metadata.ISOSpeedRatings };
+
+            var flashValues =
+                MediaUtils.formatExifFlash(metadata.Flash).map((elem) => {
+                  return navigator.mozL10n.get(elem);
```

Comment:
Is this call to get() why this file had to be listed in xfail.list?  Can you figure out another way to do it? Like putting the get() call into MediaUtils or something?